fundingmanager_test.go

```diff
@@ -347,20 +390,24 @@ func setupFundingManagers(t *testing.T, shutdownChannel chan struct{}) (*testNod
 	}
 
 	bob := &testNode{
-		privKey:      bobPrivKey,
-		msgChan:      bobMsgChan,
-		announceChan: bobAnnounceChan,
-		publTxChan:   bobPublTxChan,
-		fundingMgr:   bobFundingMgr,
-		mockNotifier: bobMockNotifier,
-		testDir:      bobTestDir,
+		privKey:         bobPrivKey,
+		msgChan:         bobMsgChan,
+		announceChan:    bobAnnounceChan,
+		arbiterChan:     bobArbiterChan,
+		publTxChan:      bobPublTxChan,
+		fundingMgr:      bobFundingMgr,
+		peer:            bobPeer,
+		mockNotifier:    bobMockNotifier,
+		testDir:         bobTestDir,
+		shutdownChannel: bobShutdownChannel,
 	}
 
 	return alice, bob
 }
 
-func tearDownFundingManagers(t *testing.T, a, b *testNode, shutdownChannel chan struct{}) {
-	close(shutdownChannel)
+func tearDownFundingManagers(t *testing.T, a, b *testNode) {
+	close(a.shutdownChannel)
+	close(b.shutdownChannel)
```

Comment:
Perhaps a wait group should be introduced to ensure that the to test nodes are _actually_ fully shutdown before proceeding?

Reply:
I don't think it is strictly necessary. The shutdownChannel is used to make sure mocked methods like `SendToPeer` returns if they are waiting for a channel send at the moment of shutdown. Immediately after closing the shutdownChannel, we call `fundingMngr.Stop()`, which will wait for all the go routines internal to fundingManager to stop. Closing the shutdown channel makes sure this `Stop()` never returns.